src/App.tsx

```diff
@@ -1,55 +1,88 @@
+
+  const handleInputBlur = () => {
+    setTimeout(() => {
+      setIsFocused(false);
+    }, 200);
+  };
+
+  const handleInputFocus = () => {
+    clearTimeout(0);
+    setIsFocused(true);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const handleSuggestionClick = (person: Person) => {
+    setInputValue(person.name);
+    setSelectedPerson(person);
+    setIsFocused(false);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter((person) => {
+      return person.name.toLowerCase().includes(inputValue.toLowerCase());
+    });
+  }, [inputValue]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={`dropdown ${isFocused ? 'is-active' : ''}`}>
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={inputValue}
+            onChange={handleInputChange}
+            onFocus={handleInputFocus}
+            onBlur={handleInputBlur}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
+          {(isFocused || inputValue === '') && (
+            filteredPeople.length > 0 ? (
+              filteredPeople.map(person => (
+                <button
+                  type="button"
+                  className="dropdown-content"
+                  key={person.name}
+                  onClick={() => handleSuggestionClick(person)}
+                >
+                  <div className="dropdown-item">
+                    <p className="has-text-link">
```

Comment:
You should do className="has-text-link" for man and className="has-text-danger" for women. In your dropdown all people have a blue color, but girls should to have red, boys should to have blue

Reply:
And also check linter mistakes in your code pls